src/node_file.cc

```diff
@@ -1061,6 +1086,8 @@ static void FTruncate(const FunctionCallbackInfo<Value>& args) {
 
 static void Fdatasync(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  if (!env->access_control()->has_permission(AccessControl::fsWrite))
```

Comment:
This is debatable. `fdatasync` flushes changes that were made earlier. Is this just here because it would be weird otherwise?

Reply:
I agree, this is debatable. :) I wasn’t sure whether one could get information out of possible error conditions, so I went for consistency.